### PR TITLE
feat(transformers): provide a flag to disable inlining views

### PR DESCRIPTION
**Pull request by TedSander**
Add a flag to allow a user to disable inlining css/html content into the views.

[Review on Reviewable](https://reviewable.io/reviews/angular/angular/2658)

### CODE REVIEW

**Comment by TedSander** [comment]
@jakemac53 what do you think of this? Isn't urgent. Just a nice to have for environments that don't want to inline html. Example when they want to use http 2.0 server push.

**Comment by jakemac53** [comment]
It should be fine, we can convert all urls to package: urls in the transformer and then we don't need mirrors in the deployed version.

**Comment by TedSander** [comment]
Cool. Then should we review it and check it in before too much drift?

**Comment by jakemac53** [comment]
Oh, somehow I missed that you actually had a cl for this :). Reviewing it now...

File: modules/angular2/src/transform/directive_processor/visitors.dart [hunk 1]
```diff
@@ -258,32 +260,34 @@ class AnnotationsTransformVisitor extends ToSourceVisitor {
...
-        return null;
-      } else {
-        logger.warning('template url is not a String $url');
-      }
-    } else if (keyString == 'styleUrls') {
-      // Inline the styleUrls
-      var urls = node.expression.accept(_evaluator);
-      writer.print('styles: const [');
-      for (var url in urls) {
+    if (_inlineViews) {
```
Comment (jakemac53) [inline modules/angular2/src/transform/directive_processor/visitors.dart]: It would probably be good here to validate that a package: url was passed, unless we add in logic to automatically convert relative/absolute urls to package: urls.

Comment (TedSander) [inline modules/angular2/src/transform/directive_processor/visitors.dart]: We are going to need the logic to automatically convert relative/absolute urls to package:urls. Also need to support web assets which we aren't doing currently. I'd rather do those in separate pull requests.

Comment (jakemac53) [inline modules/angular2/src/transform/directive_processor/visitors.dart]: Ok, can you file a bug for converting urls? I just want to make sure that doesn't fall through the cracks.

Comment (TedSander) [inline modules/angular2/src/transform/directive_processor/visitors.dart]: Done #2733

**Comment by tbosch** [comment]
Please rebase ontop of master and merge...

**Comment by TedSander** [comment]
Rebased and merged to master

---







Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/angular/angular/2658)

**Comment by munificent** [comment]
Has this actually been merged in yet?

**Comment by TedSander** [comment]
No I don't think so... I'll ping tbosch directly tomorrow.

**Comment by TedSander** [comment]
@munificent This is merged
